content/docs/functions/json_array_elements.md

```diff
@@ -0,0 +1,356 @@
+| 1  | T-Shirt | {"sizes": ["S", "M", "L", "XL"], "colors": ["Red", "Blue", "Green"]}
+| 2  | Hoodie  | {"sizes": ["XS", "S", "M", "L", "XL"], "colors": ["Black", "Gray"]}
+| 3  | Dress   | {"sizes": ["S", "M", "L"], "colors": ["Pink", "Purple", "Black"]}
+| 4  | Jeans   | {"sizes": ["28", "30", "32", "34"], "colors": ["Blue", "Black"]}
+| 5  | Jacket  | {"sizes": ["S", "M", "L", "XL"], "colors": ["Black", "Brown", "Navy"]}
+
+
+<!--
+```sql
+CREATE TABLE products (
+ id INTEGER PRIMARY KEY,
+ name TEXT,
+ details JSON
+);
+
+
+INSERT INTO products (id, name, details) VALUES
+ (1, 'T-Shirt', '{"sizes": ["S", "M", "L", "XL"], "colors": ["Red", "Blue", "Green"]}'),
+ (2, 'Hoodie', '{"sizes": ["XS", "S", "M", "L", "XL"], "colors": ["Black", "Gray"]}'),
+ (3, 'Dress', '{"sizes": ["S", "M", "L"], "colors": ["Pink", "Purple", "Black"]}'),
+ (4, 'Jeans', '{"sizes": ["28", "30", "32", "34"], "colors": ["Blue", "Black"]}'),
+ (5, 'Jacket', '{"sizes": ["S", "M", "L", "XL"], "colors": ["Black", "Brown", "Navy"]}');
+```
+-->
+
+
+`json_array_elements` is used to get all the combinations of size and color for a specific product:
+
+
+```sql
+SELECT
+ id,
+ name,
+ size,
+ color
+FROM products AS p,
+ json_array_elements(p.details -> 'sizes') AS size,
+ json_array_elements(p.details -> 'colors') AS color
+WHERE name = 'T-Shirt';
+```
+
+
+This returns:
+
+
+| id |  name   | size | color  |
+|----|---------|------|--------|
+| 1  | T-Shirt | "S"  | "Red"  |
+| 1  | T-Shirt | "S"  | "Blue" |
+| 1  | T-Shirt | "S"  | "Green"|
+| 1  | T-Shirt | "M"  | "Red"  |
+| 1  | T-Shirt | "M"  | "Blue" |
+| 1  | T-Shirt | "M"  | "Green"|
+| 1  | T-Shirt | "L"  | "Red"  |
+| 1  | T-Shirt | "L"  | "Blue" |
+| 1  | T-Shirt | "L"  | "Green"|
+| 1  | T-Shirt | "XL" | "Red"  |
+| 1  | T-Shirt | "XL" | "Blue" |
+| 1  | T-Shirt | "XL" | "Green"|
+
+
+## Filtering
+
+
+You can use the `json_array_elements` function to extract the sizes from the JSON data and then filter the products based on a specific color (or size):
+
+
+```sql
+SELECT *
+FROM products
+WHERE 'Blue' IN (
+    SELECT json_array_elements_text(details->'colors')
+);
+```
+
+
+This returns:
+
+
+| id |   name   |                               details                                |
+|----|----------|------------------------------------------------------------------------|
+|  1 | T-Shirt  | {"sizes": ["S", "M", "L", "XL"], "colors": ["Red", "Blue", "Green"]} |
+|  4 | Jeans    | {"sizes": ["28", "30", "32", "34"], "colors": ["Blue", "Black"]}     |
+
+
+
+
+## Handling null
+
+
+Update the table to insert another product (Socks) with one of the values in the `sizes` as null:
+
+
+**products**
+
+
+| id |  name   |                                 details                                 |
+|----|---------|-------------------------------------------------------------------------|
+|  6 | Socks   | {"sizes": ["S", null, "L", "XL"], "colors": ["White", "Black", "Gray"]}|
+
+
+<!--
```

Comment:
These code comments cause the build to fail. JSX comment tags can be used instead:
{ /* Comment here */ }